### ISSUE TITLE
[Proposal] Collections: second() and third() macros

### ISSUE THREAD

**Issue by clementblanco** [laravel/ideas#1411]
Similar to `collect([1, 2, 3, 4, 5])->first();` and `collect([1, 2, 3, 4, 5])->last();`.

```php
collect([1, 2, 3, 4, 5])->second();
// 2 (int)
```
```php
collect([1, 2, 3, 4, 5])->third();
// 3 (int)
```

Similar to `first()` and `last()`, those would return `null` if no item is found at the requested position.

This could be particularly useful, especially when testing and you end up have a collection of items you are using within your test.

For the time being I'm using `get()` which accepts `0` based indexes but it is not as fluent to read in plain English as "first", "second" or "last".

**Comment by deleugpn** [comment]
RoR also offer 42th

**Comment by clementblanco** [comment]
Haha yeah true

**Comment by CupOfTea696** [comment]
Adding an `nth()` macro might be something to consider, but if you're gonna add `second()` and `third()`, where does it end?

**Comment by CupOfTea696** [comment]
Oh yeah, didn't realise there was already an `nth()` method. It has a totally different functionality, though. This one could be named `getNth()`.
